.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+# SPDX-FileCopyrightText: Copyright 2024 Siemens AG
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: release browser extension
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+
+# Sets permissions of the GITHUB_TOKEN to checkout the repository
+permissions:
+  contents: write
+
+env:
+  WEB_EXT_VERS: 8.2.0
+
+jobs:
+  release-xpi:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v4
+
+      - name: install dependencies
+        run: |
+          sudo apt update && sudo apt install -y make git zip
+
+      - run: make package
+
+      - name: sign Firefox extension by Mozilla
+        run: |
+          npx web-ext@${{ env.WEB_EXT_VERS }} sign \
+            --channel unlisted \
+            --approval-timeout 900000 \
+            --api-key ${{ secrets.AMO_API_KEY }} \
+            --api-secret ${{ secrets.AMO_API_SECRET }} \
+            --source-dir build/firefox \
+            --artifacts-dir build
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: firefox-signed-xpi
+          path: |
+            build/linux_entra_sso-*.xpi
+
+      - name: create release
+        uses: softprops/action-gh-release@v2
+        with:
+          files: |
+            build/linux_entra_sso-*.xpi
+          token: ${{ secrets.GITHUB_TOKEN }}
+          body: |
+            Release of version ${{ github.ref_name }}
+          tag_name: ${{ github.ref_name }}
+          draft: false
+          prerelease: true
+
+      - name: create update manifest for Firefox
+        run: |
+          VERSION=$(echo ${{ github.ref_name }} | cut -c 2-)
+          DIGEST="sha256:$(sha256sum build/linux_entra_sso-${VERSION}.xpi | cut -d ' ' -f 1)"
+          LINK="https://github.com/siemens/linux-entra-sso/releases/download/v${VERSION}/linux_entra_sso-${VERSION}.xpi"
+          jq --arg version "${VERSION}" --arg digest "${DIGEST}" --arg link "${LINK}" \
+            '."addons"."linux-entra-sso@example.com"."updates" += [{"version":$version, "update_link":$link, "update_hash":$digest}]' \
+            .pages/firefox/updates.json \
+            > .pages/firefox/updates.json.tmp && mv .pages/firefox/updates.json.tmp .pages/firefox/updates.json
+
+      - name: commit update manifest for Firefox
+        uses: stefanzweifel/git-auto-commit-action@v5
```

Comment:
Shouldn't we better pin this dependency to a specific version?

Reply:
In the github ecosystem, the versions are usually just pinned to a tag, but not to a specific commit. By that, I want to keep it like that.